src/main/java/codechicken/enderstorage/container/ContainerEnderItemStorage.java

```diff
@@ -93,4 +90,23 @@ public void onContainerClosed(EntityPlayer entityplayer) {
         super.onContainerClosed(entityplayer);
         chestInv.closeInventory();
     }
+
+    @Override
+    public boolean canInteractWith(EntityPlayer entityplayer) {
+        return chestInv.isUsableByPlayer(entityplayer);
+    }
+
+    @ChestContainer.RowSizeCallback
+    public int getRowSize() {
+        switch (chestInv.getSize()) {
+            case 0:
+                return 3;
+            case 1:
+                return 9;
+            case 2:
+                return 9;
```

Comment:
This can also be cleaned up to something like this
```java
switch (...) {
    case 0:
        return 3;
    case 1:
    case 2:
        return 9;
    default:
        throw ...;
}
```